src/services/DockerService.ts

```diff
@@ -136,6 +141,54 @@ export class DockerService implements IService{
         }
         return false;
     }
+
+    public async checkDockerResources(isMultiNodeMode: boolean) {
+      this.logger.info('Checking docker resources...', this.serviceName);
+      const resultDockerInfoCommand = await shell.exec(
+        'docker system info --format=json',
+        { silent: true }
+      );
+      
+      const systemInfoJson = JSON.parse(resultDockerInfoCommand.stdout);
+      const dockerMemory = Math.round(systemInfoJson['MemTotal'] / Math.pow(1024, 3));
+      const dockerCPUs = systemInfoJson['NCPU'];
+
+      return this.checkMemoryResources(dockerMemory, isMultiNodeMode) &&
+      this.checkCPUResources(dockerCPUs);
+    }
+
+    private checkMemoryResources(dockerMemory: number, isMultiNodeMode: boolean) {
+      if (dockerMemory >= MIN_MEMORY_SINGLE_MODE && dockerMemory < RECOMMENDED_MEMORY_SINGLE_MODE && !isMultiNodeMode) {
+        this.logger.warn(`Your docker memory resources are ${dockerMemory.toFixed(2)}GB, which may cause unstable behaviour. Set to at least ${RECOMMENDED_MEMORY_SINGLE_MODE}GB`, this.serviceName);
+        return true;
+      } else if (dockerMemory < MIN_MEMORY_SINGLE_MODE && !isMultiNodeMode) {
+        this.handleMemoryError(dockerMemory, isMultiNodeMode);
+        return false;
+      } else if(dockerMemory < MIN_MEMORY_MULTI_MODE && isMultiNodeMode) {
+        this.handleMemoryError(dockerMemory, isMultiNodeMode);
+        return false;
+      }
+
+      return true;
+    }
+
+    private checkCPUResources(dockerCPUs: number) {
+      if(dockerCPUs >= MIN_CPUS && dockerCPUs < RECOMMENDED_CPUS && !process.env.CI) {
+        this.logger.warn(`Your docker CPU resources are set to ${dockerCPUs}, which may cause unstable behaviour. Set to at least ${RECOMMENDED_CPUS} CPUs`, this.serviceName);
+        return true;
+      } else if (dockerCPUs < MIN_CPUS && !process.env.CI) {
+        this.logger.error(`Your docker CPU resources are set to ${dockerCPUs}. This is not enough, set to at least ${RECOMMENDED_CPUS} CPUs`, this.serviceName);
+        return false;
+      }
+
+      return true;
+    }
+
+    private handleMemoryError(dockerMemory: number, isMultiNodeMode: boolean) {
+      const recommendedMemory = isMultiNodeMode ? MIN_MEMORY_MULTI_MODE : MIN_MEMORY_SINGLE_MODE;
+      this.logger.error(`Your docker memory resources are set to ${dockerMemory.toFixed(2)}GB. This is not enough, set to at least ${recommendedMemory}GB`, this.serviceName);
+      process.exit(1);
```

Comment:
Won't this exit the program, before it reaches the return later ?

Reply:
solved